src/instructlab/rag/rag_configuration.py

```diff
@@ -0,0 +1,141 @@
+# Standard
+from typing import Any, Optional
+import logging
+import os
+
+# Third Party
+from pydantic import BaseModel, ConfigDict
+import click
+
+# First Party
+from instructlab.configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+
+def rag_options(command):
```

Comment:
will this wrapper apply these to any `ilab` command using the @rag_options? I would really prefer for ilab commands directly using rag and allowing users to configure RAG, that they have their own validated set of options that read from the config. 

A second option I would be ok with, is if these `rag_options` use the `cls=clickext.ConfigOption` class to read their defaults from the config file.

Reply:
> will this wrapper apply these to any ilab command using the @rag_options?
This was meant to be used only in the chat command, where it makes sense to add these options.

> I would really prefer for ilab commands directly using rag and allowing users to configure RAG, that they have their own validated set of options that read from the config
👍